src/main/java/core/basesyntax/db/StockDaoStorageImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.db;
+
+import core.basesyntax.storage.Storage;
+import java.util.Set;
+
+public class StockDaoStorageImpl implements StockDao {
+
+    @Override
+    public void set(String product, Integer amount) {
+        Storage.stock.put(product, amount);
+    }
+
+    @Override
+    public Integer get(String product) {
+        return Storage.stock.get(product);
+    }
+
+    @Override
+    public void decrease(String product, Integer amount) {
+        if (contain(product)) {
+            Storage.stock.put(product, get(product) - amount);

```

Comment:
Consider adding a check to ensure that the resulting stock amount is not negative after decreasing. This will prevent logical errors where stock becomes negative, which is usually not desired in inventory systems.